fabfile/tasks.py

```diff
@@ -0,0 +1,147 @@
+            if i
+        ]
+        result['flake8_violations'] = [
+            i
+            for i in flake8(path_to_file).split("\n")
+            if i
+        ]
+        result['pylint_violations'] = [
+            i
+            for i in pylint(path_to_file).split("\n")
+            if all([
+                i,
+                not i.startswith(' '),
+                not i.startswith('\r'),
+                not i.startswith('-'),
+                not i.startswith('Y'),
+                not i.startswith('*'),
+                not i.startswith('Using config file'),
+            ])
+        ]
+        result['path_to_file'] = path_to_file
+        result['total_violations'] = sum(
+            [
+                len(result['pycodestyle_violations']),
+                len(result['flake8_violations']),
+                len(result['pylint_violations']),
+            ]
+        )
+        results.append(result)
+    return results
+
+
+def print_item(item, verbose, details):
+    """Print an item with the appropriate verbosity / detail"""
+    if verbose:
+        line = "{0} {1} {2} {3} {4}".format(
+            item['total_violations'],
+            len(item['pycodestyle_violations']),
+            len(item['flake8_violations']),
+            len(item['pylint_violations']),
+            item['path_to_file'],
+        )
+    else:
+        line = item['path_to_file']
+    print line
+
+    if details:
+        print "pycodestyle:"
+        for detail in flatten(item['pycodestyle_violations']):
+            print detail
+        print
+
+        print "flake8:"
+        for detail in flatten(item['flake8_violations']):
+            print detail
+        print
+
+        print "pylint:"
+        for detail in flatten(item['pylint_violations']):
+            print detail
+        print
+
+
+@task
+def code_quality(top=10, verbose=True, details=False, fix=False, filename=None):
+    """
+    Check code quality.
+
+    By default this command will analyse each Python file in the project with a variety of tools and display the count
+    or details of the violations discovered, sorted by most iolations first.
+
+    Default usage:
+
+    $ fab -H localhost code_quality
+
+    Options:
+        top: Display / fix only the top violating files, a value of 0 will display / fix all files
+        verbose: Display a header and the counts. Without this you just get the filenames in order
+        details: Display the violations one per line after the count/file summary
+        fix: Run autopep8 on the displayed file(s)
+        filename: Rather than analysing all files and displaying / fixing the top N, just analyse / diaply / fix the
+        specified file
+
+    Intended to be temporary until we have improved code quality and have safeguards to maintain it in place.
+
+    """
+
+    verbose = coerce_bool(verbose)
+    details = coerce_bool(details)
+    fix = coerce_bool(fix)
+    end = int(top) if int(top) else -1
```

Comment:
`end = int(top) or -1`

Reply:
I want 0 to mean no slicing.

Reply:
it's just a shorter record of the same expression.

Reply:
Ah right, yes indeed. Amazing what a second pair of eyes can do! Fix incoming, along with fixing a bug where end was ignored if `fix` was `True`.